leakcanary-android-instrumentation/src/main/java/com/squareup/leakcanary/InstrumentationLeakDetector.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.leakcanary;
+
+import android.app.Application;
+import android.app.Instrumentation;
+import android.content.Context;
+import android.os.Debug;
+import android.os.SystemClock;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+
+public final class InstrumentationLeakDetector {
+
+  /**
+   * @return a {@link} AndroidRefWatcherBuilder that will create a {@link RefWatcher} suitable for
+   * instrumentation tests. This {@link RefWatcher} will never trigger a heap dump.
+   */
+  public static AndroidRefWatcherBuilder instrumentationRefWatcher(Application application) {
+    return LeakCanary.refWatcher(application)
+        .watchExecutor(new WatchExecutor() {
+          // Storing weak refs to ensure they make it to the queue.
+          final List<Retryable> trackedReferences = new CopyOnWriteArrayList<>();
+
+          @Override public void execute(Retryable retryable) {
+            trackedReferences.add(retryable);
+          }
+        });
+  }
+
+  public InstrumentationLeakResults detectLeaks() {
```

Comment:
should this be public? should it only be called from `FailTestOnLeakRunListener`? or is the idea that you can create your own FailTestOnLeakRunListener by using this method?